modules/sdk-coin-atom/src/atom.ts

```diff
@@ -226,6 +256,221 @@ export class Atom extends BaseCoin {
+      const sendMessage: SendMessage[] = [
+        {
+          fromAddress: senderAddress,
+          toAddress: params.recoveryDestination,
+          amount: amount,
+        },
+      ];
+      const gasBudget: FeeData = {
+        amount: [{ denom: 'uatom', amount: '500' }],
+        gasLimit: 200000,
+      };
+
+      // Step 7: Build the unsigned tx
+      const txnBuilder = this.getBuilder().getTransferBuilder();
+      txnBuilder
+        .messages(sendMessage)
+        .gasBudget(gasBudget)
+        .publicKey(publicKey)
+        .accountNumber(Number(accountNumber))
+        .chainId(chainId);
+      const unsignedTransaction = (await txnBuilder.build()) as Transaction;
+      let serializedTx = unsignedTransaction.toBroadcastFormat();
+      const signableHex = unsignedTransaction.signablePayload.toString('hex');
+
+      // Step 8: Validate necessary parameters required for performing recovery
+      if (!params.userKey) {
+        throw new Error('missing userKey');
+      }
+      if (!params.backupKey) {
+        throw new Error('missing backupKey');
+      }
+      if (!params.walletPassphrase) {
+        throw new Error('missing wallet passphrase');
+      }
+
+      const userKey = params.userKey.replace(/\s/g, '');
+      const backupKey = params.backupKey.replace(/\s/g, '');
+      const [userKeyCombined, backupKeyCombined] = ((): [
+        ECDSAMethodTypes.KeyCombined | undefined,
+        ECDSAMethodTypes.KeyCombined | undefined
+      ] => {
+        const [userKeyCombined, backupKeyCombined] = this.getKeyCombinedFromTssKeyShares(
+          userKey,
+          backupKey,
+          params.walletPassphrase
+        );
+        return [userKeyCombined, backupKeyCombined];
+      })();
+
+      if (!userKeyCombined || !backupKeyCombined) {
+        throw new Error('Missing key combined shares for user or backup');
+      }
+
+      // Step 9: Sign the tx
+      const signature = await this.signRecoveryTSS(userKeyCombined, backupKeyCombined, signableHex);
+      const atomKeyPair = new AtomKeyPair({ pub: publicKey });
+      txnBuilder.addSignature({ pub: atomKeyPair.getKeys().pub }, Buffer.from(signature.r + signature.s, 'hex'));
+      const signedTransaction = await txnBuilder.build();
+      serializedTx = signedTransaction.toBroadcastFormat();
+
+      return { serializedTx: serializedTx, scanIndex: i };
+    }
+    throw new Error('Did not find an address with funds to recover');
+  }
+
+  /**
+   * Get balance from public node
+   */
+  protected async getBalanceFromNode(senderAddress: string): Promise<request.Response> {
+    const nodeUrl = this.getPublicNodeUrl();
+    const getBalancePath = '/cosmos/bank/v1beta1/balances/';
+    const fullEndpoint = nodeUrl + getBalancePath + senderAddress;
+    try {
+      return await request.get(fullEndpoint).send();
+    } catch (e) {
+      console.debug(e);
+    }
+    throw new Error(`Unable to call endpoint ${getBalancePath + senderAddress} from node: ${nodeUrl}`);
```

Comment:
Will this response directly give you the balance, if not, Try to get the balance and return only the balance so that the logic is abstracted within this function

Reply:
This is actually done one step above in the chain. So the flow goes like this:

* Recover function calls `getAccountBalance(address)` and expects a balance string as response.
* getAccountBalance(address) calls another helper function called `getBalanceFromNode` to actually make the call to the full node. Once the helper responds, `getAccountBalance(address)` checks if the response is a 200 or otherwise; if it is a 200 indeed, then it parses the response here since the helper's job is to return a response alone
* The helper meanwhile constructs the REST API call based of the parameters that were provided and simply makes the call and gives back the response

This flow applies for all calls we make to the full node to fetch the different parameters. We could probably do this in one single call instead but this way, we can abstract whats needed/not needed